fbuapp/View Controllers/HomeViewController.m

```diff
@@ -0,0 +1,163 @@
+//
+//  HomeViewController.m
+//  fbuapp
+//
+//  Created by danielavila on 7/19/19.
+//  Copyright © 2019 eazheng. All rights reserved.
+//
+
+#import "HomeViewController.h"
+#import "PostCell.h"
+#import "Parse/Parse.h"
+#import "Post.h"
+#import <CoreLocation/CoreLocation.h>
+#import "UIImageView+AFNetworking.h"
+#import "DateTools.h"
+
+static NSString *kTableViewPostCell = @"PostCell";
+
+
+@interface HomeViewController () <UITableViewDataSource, UITableViewDelegate, CLLocationManagerDelegate, UIScrollViewDelegate>
+@property (strong, nonatomic) NSArray * posts;
+@property (nonatomic, strong) UIRefreshControl *refreshControl;
+@property (strong, nonatomic) CLLocation * currentLocation;
+@property (nonatomic,strong) CLLocationManager *locationManager;
+
+@end
+
+@implementation HomeViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    
+    self.locationManager = [[CLLocationManager alloc] init];
+    self.locationManager.delegate = self;
+    self.locationManager.desiredAccuracy = kCLLocationAccuracyBest;
+    [self.locationManager requestWhenInUseAuthorization];
+    [self.locationManager startUpdatingLocation];
+    
+    [self.tableView registerNib:[UINib nibWithNibName:kTableViewPostCell bundle:nil] forCellReuseIdentifier:kTableViewPostCell];
+    
+    self.tableView.dataSource = self;
+    self.tableView.delegate = self;
+    
+    [self fetchPosts];
+    
+    self.refreshControl = [[UIRefreshControl alloc] init];
+    [self.refreshControl addTarget:self action:@selector(fetchPosts) forControlEvents:UIControlEventValueChanged];
+    [self.tableView addSubview:self.refreshControl];
+}
+
+- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section{
+    return self.posts.count;
+}
+
+- (void)locationManager:(CLLocationManager *)manager didUpdateLocations:(NSArray *)locations
+{
+    self.currentLocation = [locations lastObject];
+}
+
+-(void)locationManager:(CLLocationManager *)manager didFailWithError:(NSError *)error{
+    UIAlertController* alert = [UIAlertController alertControllerWithTitle:@"Location Services Error"
+                                                                   message:[NSString stringWithFormat:@"%@", error.localizedDescription]
```

Comment:
we should create a helper method to show UIAlert so we can just call the function here.

Reply:
That's a great suggestion. Can you please add one @danthecodingman?